### PR TITLE
Rendered content for activity discussion

### LINKED ISSUE #2624: Forum related activities UI issue when forum component is deactivated
**Describe the bug**
When the forum component is deactivated. Its activities like: forum replies and new discussion activities has UI issue.

**To Reproduce**
1. Go to Forum
2. Make reply on any discussion, Create a new discussion
3. Now, Deactivate the forum component
4. Check those activities UI now.

**Expected behavior**
If the forum component is deactivated then related activities should not break.

**Screenshots**
https://ibb.co/8dfJpZG

**Environment**
My localhost on release branch

### LINKED ISSUE #2640: Discussion reply activity showing only empty border no content in activity pages.
**Describe the bug**
In the activity pages when you have media data but no content for reply activity then it will show the empty border.
 
**To Reproduce**
1. Create discussion.
2. Create reply according the discussion with media data and no content (text).
3. Disabled the forum from the back-end settings.
4. Navigate the activity page.
5. Now you get the discussion reply activity with only empty border no content. 

**Expected behavior**
We should prevent render the reply activity when its content is empty and the forum is disabled.

**Screenshots**
https://prnt.sc/1dhv43d

**Environment**
Local server.

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.

### PR DESCRIPTION
**Pull request by asaquzzamanbb**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2624.
Fixes #2640.

### How to test the changes in this Pull Request:

1. Go to Forum
2. Make reply on any discussion, Create a new discussion
3. Now, Deactivate the forum component
4. Check those activities UI now.

### Proof Screenshots or Video

https://prntscr.com/1d15z6z

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-activity/bp-activity-filters.php [hunk 1]
```diff
@@ -159,8 +159,69 @@
 // Filter check content empty or not for the media, document and GIF data.
 add_filter( 'bb_is_activity_content_empty', 'bb_check_is_activity_content_empty' );
 
+// Filter activity discussion content.
+add_filter( 'bp_get_activity_content_body', 'bb_before_activity_content', 10, 2 );
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:163]:
```suggestion
add_filter( 'bp_get_activity_content_body', 'bb_disabled_forum_render_activity_content', 10, 2 );
```

File: src/bp-activity/bp-activity-filters.php [hunk 2]
```diff
@@ -159,8 +159,69 @@
...
+ *
+ * @param string $content  Activit content.
+ * @param object $activity Activit data.
+ *
+ * @uses bbp_get_reply()           Get reply post data.
+ * @uses bbp_get_topic_permalink() Get discussion permalink.
+ *
+ * @return string
+ */
+function bb_before_activity_content( $content, $activity ) {
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:180]:
```suggestion
function bb_disabled_forum_render_activity_content( $content, $activity ) {
```

File: src/bp-activity/bp-activity-filters.php [hunk 3]
```diff
@@ -159,8 +159,69 @@
...
+
+	// When the activity type does not match with the topic or reply.
+	if ( ! in_array( $activity->type, array( 'bbp_topic_create', 'bbp_reply_create' ), true ) ) {
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:183]:
```suggestion
	if ( ! bp_is_active( 'forums' ) && in_array( $activity->type, array( 'bbp_topic_create', 'bbp_reply_create' ), true ) ) {
```

File: src/bp-activity/bp-activity-filters.php [hunk 4]
```diff
@@ -159,8 +159,69 @@
...
+		return $content;
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:184]:
```suggestion
		return sprintf( '<div class="bb-content-inr-wrap">%1$s</div>', $content );
```

File: src/bp-activity/bp-activity-filters.php [hunk 5]
```diff
@@ -159,8 +159,69 @@
...
+		}
+	}
+
+	// Topic.
+	$topic_permalink = bbp_get_topic_permalink( $topic_id );
+	$topic_title     = get_post_field( 'post_title', $topic_id, 'raw' );
+	$reply_to_text   = ( ! empty( $topic->ID ) && bbp_is_reply( $topic->ID ) ) ? sprintf( '<span class="bb-reply-lable">%1$s</span>', esc_html__( 'Reply to', 'buddyboss' ) ) : '';
+	$content         = sprintf( '<p class = "activity-discussion-title-wrap"><a href="%1$s">%2$s %3$s</a></p> %4$s', esc_url( $topic_permalink ), $reply_to_text, $topic_title, $rendered_content );
+
+	return $content;
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:223]: Remove this unnecessary code

File: src/bp-forums/activity.php [hunk 6]
```diff
@@ -169,12 +169,8 @@ private function setup_filters() {
 			// Filter for single topic.
 			add_filter( 'bbp_is_single_topic', array( $this, 'set_single_topic' ) );
 
-			// Filter discussion.
-			add_filter( 'bp_get_activity_content_body', array( $this, 'before_activity_content' ), 10, 2 );
```
Comment (chetansatasiya) [inline src/bp-forums/activity.php:173]: Please don't remove this filter.

File: src/bp-forums/activity.php [hunk 7]
```diff
@@ -348,60 +344,6 @@ function activity_get_notification_permalink( $link, $item_id, $secondary_item_i
...
-
-			// Topic.
-			$topic_permalink = bbp_get_topic_permalink( $topic_id );
-			$topic_title     = get_post_field( 'post_title', $topic_id, 'raw' );
-			$reply_to_text   = ( ! empty( $topic->ID ) && bbp_is_reply( $topic->ID ) ) ? sprintf( '<span class="bb-reply-lable">%1$s</span>', esc_html__( 'Reply to', 'buddyboss' ) ) : '';
-			$content         = sprintf( '<p class = "activity-discussion-title-wrap"><a href="%1$s">%2$s %3$s</a></p> <div class="bb-content-inr-wrap">%4$s</div>', esc_url( $topic_permalink ), $reply_to_text, $topic_title, $content );
-
-			return $content;
-		}
-
```
Comment (chetansatasiya) [inline src/bp-forums/activity.php:404]: Please don't remove this function

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Please do the requested changes

File: src/bp-forums/activity.php (hunk 7 above)
Comment (asaquzzamanbb) [reply src/bp-forums/activity.php]: Back to the previous one.
https://github.com/buddyboss/buddyboss-platform/pull/2635/commits/7593d31b8fde819594390a94b715746fd0abbee1

File: src/bp-forums/activity.php (hunk 6 above)
Comment (asaquzzamanbb) [reply src/bp-forums/activity.php]: Back to the previous one.
https://github.com/buddyboss/buddyboss-platform/pull/2635/commits/7593d31b8fde819594390a94b715746fd0abbee1

File: src/bp-activity/bp-activity-filters.php (hunk 5 above)
Comment (asaquzzamanbb) [reply src/bp-activity/bp-activity-filters.php]: Removed here
https://github.com/buddyboss/buddyboss-platform/pull/2635/commits/ec328a3ae9ef0e12a87232931f405efc3a444966

File: src/bp-activity/bp-activity-filters.php [hunk 8]
```diff
@@ -159,8 +159,33 @@
...
+function bb_disabled_forum_render_activity_content( $content, $activity ) {
+
+	// When forum is disabled and the activity type match with the topic or reply.
+	if ( ! bp_is_active( 'forums' ) && in_array( $activity->type, array( 'bbp_topic_create', 'bbp_reply_create' ), true ) ) {
+		return sprintf( '<div class="bb-content-inr-wrap">%1$s</div>', $content );
+	}
+
+	return $content;
+}
 /**
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:189]: Add line break here

File: src/bp-forums/activity.php [hunk 9]
```diff
@@ -1079,4 +1078,4 @@ public function set_single_topic( $single_topic ) {
 			return $single_topic;
 		}
 	}
-endif;
+endif;
```
Comment (chetansatasiya) [inline src/bp-forums/activity.php:1081]: Add line break here

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
do the changes as requested

File: src/bp-activity/bp-activity-filters.php (hunk 8 above)
Comment (asaquzzamanbb) [reply src/bp-activity/bp-activity-filters.php]: Done
https://github.com/buddyboss/buddyboss-platform/pull/2635/commits/4fd55ff54a198875befd6c98b1f99b4cb2d70b0c

File: src/bp-forums/activity.php (hunk 9 above)
Comment (asaquzzamanbb) [reply src/bp-forums/activity.php]: Changes done
https://github.com/buddyboss/buddyboss-platform/pull/2635/commits/20624fc82b2f93ff3bb5cfb783f17a388fa590e0

**Review by chetansatasiya — DISMISSED** [review]
LGTM 🚀 

Merge in dev and add in testing

**Comment by KartikSuthar** [comment]
@sourabhmatolia 
As discussed. We have removed forums activity while the forum component is disabled.
Same for the Group as well.

CC: @chetansatasiya

**Review by chetansatasiya — APPROVED** [review]

**Comment by mohit-buddyboss** [comment]
@KartikSuthar 

Tested this changes on dev branch of the platform.
Tested this by creating Individual Forum related activity, Public group related discussion and update activity and also Private group related discussion and update activity. 


**Scenario 1 : ON deactivating FORUM component**
1) Now on deactivating Forum component all activities related to Individual Forums and Group Forum discussions and replies will get hide from activity page and group feed tab. - **PASS**
2) Discussion tab will not appear in Groups. - **PASS**
3) Forum directory page will not appear. - **PASS**
4) Under moderation Forums option will not appear - **PASS**

**Scenario 2: On deactivating Social Groups component:**

1) All group related activities must will get hide from news feed page - **PASS**
2) Groups Directory page must not appear -**PASS**
3) Under moderation Group reporting option must not appear - **PASS**
4) On photos directory page group related media must not appear - **PASS**
5) On documents page group related documents must not appear -**PASS**

6) On videos directory page group related videos must not appear - **FAIL** 
Created separate issue for this : https://github.com/buddyboss/buddyboss-platform/issues/2654

7) If group component is disable but forum component is active then Public group discussions will be treated as Individual Public Forum discussion and creating any new reply or discussion will create it's new activity as a Individual Forum. -**PASS**

Video : https://loom.com/share/4ef0b8180a5045adb009854a780cb821

Related issues are resolve now and PR is ready for release.

cc @sourabhmatolia @chetansatasiya

**Pull request merged**